extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,152 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+export const VALIDATION_OBJECTS = {
+  'aggregateRating': {
+    'ratingValue': [validateRequired, validateNumber],
+    'reviewCount': [validateRequired, validateNumber],
+    'reviewUrl': [validateRequired, validateURLs],
+  },
+  'productValidationConfig': {
+    /* Required Attrs */
+    'productUrl': [validateRequired, validateString],
+    'productId': [validateRequired, validateString],
+    'productTitle': [validateRequired, validateString],
+    'productBrand': [validateRequired, validateString],
+    'productPrice': [validateRequired, validateNumber],
+    'productImages': [validateRequired, validateURLs],
+    'productPriceCurrency': [validateRequired, validateString],
+    'aggregateRating': [validateRequired, validateObject],
+    /* Optional Attrs */
+    'productColor': [validateString],
+    'productSize': [validateString],
+    'productIcon': [validateURLs],
+    'productTagText': [validateString],
+    'reviewsData': [validateURLs],
+    'ctaText': [validateNumber],
+    'shippingText': [validateNumber],
+  },
+};
+
+const essentialFields = [
+  'productPrice',
+  'productTitle',
+  'productImages',
+  'productBrand',
+  'productId',
+];
+
 /**
  * @typedef {{
  *  items: !Array<!ShoppingConfigDataDef>,
  * }}
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Validates an Object using its field name as a key for one of the validation objects above.
+ * @param {string} field
+ * @param {?Object=} value
+ */
+function validateObject(field, value = undefined) {
+  validateConfig(value, VALIDATION_OBJECTS[field]);
+}
+
+/**
+ * Validates if a required field exists for shopping config attributes
+ * @param {string} field
+ * @param {?string=} value
+ */
+export function validateRequired(field, value = undefined) {
```

Comment:
Unset arguments always default to undefined, you don't need the param initializer here or elsewhere.

```suggestion
export function validateRequired(field, value) {
```

Reply:
Sounds good.